File: Server/Dto/Product.cs
using System;
using System.Collections.Generic;

 namespace Dto;

public partial class Product
{
    public int ProdId { get; set; }

    public string ProdName { get; set; } = null!;

    public int? CatCode { get; set; }

    public int? CompanyCode { get; set; }

    public string? Description { get; set; }

    public decimal? Price { get; set; }

    public string? Image { get; set; }

    public int? StockQty { get; set; }
    public int Count {  get; set; }

    public DateOnly? LastUpdated { get; set; }



}

File: Client/src/classes/Product.ts
import { Category } from "./Category";
// import { Company } from "./Company";
import { Company } from "./Company";
export class Product {
    constructor(public prodId?: number,
        public prodName?: String,
        public catCode:Category=new Category(),
        public companyCode:Company=new Company(),
        public description?: String,
        public price?: number,
        public image?: string,
        public stockQty?: number,
        public lastUpdated?: Date,
    public startShop?:Boolean,
public count:number=0) { }
}
